src/openrct2/world/Scenery.cpp

```diff
@@ -281,3 +284,149 @@ int32_t wall_entry_get_door_sound(const WallSceneryEntry* wallEntry)
+        case SCENERY_TYPE_LARGE:
+            return MAX_LARGE_SCENERY_OBJECTS;
+        case SCENERY_TYPE_BANNER:
+            return MAX_BANNER_OBJECTS;
+        default:
+            return 0;
+    }
+}
+
+static SceneryEntryBase* GetSceneryEntry(ScenerySelection item)
+{
+    switch (item.SceneryType)
+    {
+        case SCENERY_TYPE_SMALL:
+            return get_small_scenery_entry(item.EntryIndex);
+        case SCENERY_TYPE_PATH_ITEM:
+            return get_footpath_item_entry(item.EntryIndex);
+        case SCENERY_TYPE_WALL:
+            return get_wall_entry(item.EntryIndex);
+        case SCENERY_TYPE_LARGE:
+            return get_large_scenery_entry(item.EntryIndex);
+        case SCENERY_TYPE_BANNER:
+            return get_banner_entry(item.EntryIndex);
+        default:
+            return nullptr;
+    }
+}
+
+bool IsSceneryItemRestricted(ScenerySelection item)
+{
+    auto it = std::find(_restrictedScenery.begin(), _restrictedScenery.end(), item);
+    return it != _restrictedScenery.end();
+}
+
+void ClearRestrictedScenery()
+{
+    _restrictedScenery.clear();
+}
+
+std::vector<ScenerySelection>& GetRestrictedScenery()
+{
+    return _restrictedScenery;
+}
+
+void RestrictAllMiscScenery()
+{
+    std::vector<ScenerySelection> nonMiscScenery;
+    for (ObjectEntryIndex i = 0; i < MAX_SCENERY_GROUP_OBJECTS; i++)
+    {
+        auto sgEntry = get_scenery_group_entry(i);
+        if (sgEntry != nullptr)
+        {
+            for (size_t j = 0; j < sgEntry->entry_count; j++)
+            {
+                nonMiscScenery.push_back(sgEntry->scenery_entries[j]);
+            }
+        }
+    }
+    for (uint8_t sceneryType = SCENERY_TYPE_SMALL; sceneryType < SCENERY_TYPE_COUNT; sceneryType++)
+    {
+        auto maxObjects = GetMaxObjectsForSceneryType(sceneryType);
+        for (ObjectEntryIndex i = 0; i < maxObjects; i++)
+        {
+            ScenerySelection sceneryItem = { sceneryType, i };
+            auto sceneryEntry = GetSceneryEntry(sceneryItem);
+            if (sceneryEntry != nullptr)
+            {
+                auto it = std::find(nonMiscScenery.begin(), nonMiscScenery.end(), sceneryItem);
+                if (it == nonMiscScenery.end())
+                {
+                    _restrictedScenery.push_back(sceneryItem);
+                }
+            }
+        }
+    }
+}
+
+ObjectType GetObjectTypeFromSceneryType(uint8_t type)
+{
+    switch (type)
+    {
+        case SCENERY_TYPE_SMALL:
+            return ObjectType::SmallScenery;
+        case SCENERY_TYPE_PATH_ITEM:
+            return ObjectType::PathBits;
+        case SCENERY_TYPE_WALL:
+            return ObjectType::Walls;
+        case SCENERY_TYPE_LARGE:
+            return ObjectType::LargeScenery;
+        case SCENERY_TYPE_BANNER:
+            return ObjectType::Banners;
+        default:
+            throw std::runtime_error("Invalid scenery type");
+    }
+}
+
+uint8_t GetSceneryTypeFromObjectType(ObjectType type)
```

Comment:
The same code also resides in the SceneryGroupObject. That code would need to be modified to use this as well.

Reply:
I dont really see the point in the SceneryType object. I'm going to look see if we can just remove it.

Reply:
Eitherway these two functions are unused so I'm removing them.